Make failed for zkc while linking

On my system `gem install zookeeper` fails with:
```bash
linking shared-object zookeeper_c.so
/home/ullrich/.gem/ruby/1.9.1/gems/zookeeper-1.4.8/ext/lib/libzookeeper_st_gem.a(hashtable.o): In function `create_hashtable':
/home/ullrich/.gem/ruby/1.9.1/gems/zookeeper-1.4.8/ext/zkc-3.4.5/c/src/hashtable/hashtable.c:51: undefined reference to `ceilf'
/home/ullrich/.gem/ruby/1.9.1/gems/zookeeper-1.4.8/ext/lib/libzookeeper_st_gem.a(hashtable.o): In function `hashtable_expand':
/home/ullrich/.gem/ruby/1.9.1/gems/zookeeper-1.4.8/ext/zkc-3.4.5/c/src/hashtable/hashtable.c:124: undefined reference to `ceilf'
collect2: ld returned 1 exit status
make: *** [zookeeper_c.so] Error 1
```

I was able to get it compiling by hand when changing the order for the linker to have `-lm` **after** `-lzookeeper_st_gem` in the Makefile.

I am unsure what can be broken here. Any Ideas?

--- a/ext/extconf.rb
+++ b/ext/extconf.rb
@@ -89,7 +89,9 @@ Dir.chdir("#{HERE}/lib") do
     end
   end
 end
-$LIBS << " -lzookeeper_st_gem"
+
+# -lm must come after lzookeeper_st_gem to ensure proper link
+$LIBS << " -lzookeeper_st_gem -lm"
 
 have_func('rb_thread_blocking_region')
 have_func('rb_thread_fd_select')

